src/commonMain/kotlin/app/cash/turbine/flow.kt

```diff
@@ -24,12 +26,41 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.channels.Channel
 import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
 import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.currentCoroutineContext
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.job
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.plus
 import kotlinx.coroutines.test.TestCoroutineScheduler
 import kotlinx.coroutines.test.UnconfinedTestDispatcher
 
+public interface TurbineTestContext<T> : ReceiveTurbine<T> {
```

Comment:
Notes on the addition of `TurbineTestContext`...

This stuff all works by creating a "turbine registry" and sticking it in the coroutine context. Great, except for `TestScope`. 

`TestScope` creates a situation in which there are "two" parent scopes: the regular `TestScope`, and the `backgroundScope`. So if `withContext` is called within `runTest`, none of its changes are applied to `backgroundScope`.

This breaks `withTurbineTimeout` inside `backgroundScope`, but... for obvious reasons, nobody noticed. 

It matters a lot more for `Flow.testIn(CoroutineScope)`. `backgroundScope` doesn't have the registry. So: what to do?

I went with the approach of creating a `TurbineTestContext` receiver that includes an override for `testIn` that does a second `withContext` on the backgroundScope. The downside of this approach is that any subsequent code that discards that context will lose this tracking.

Another approach that could work would be to modify `testIn` to have a signature of `suspend fun <T> Flow<T>.testIn(...)`. This would allow access to the calling context via `currentCoroutineContext()`. Still breaks inside `backgroundScope`, though.